# Carga de librerias

In [1]:
from pymongo import MongoClient
import time
from datetime import datetime, timedelta

client = MongoClient('localhost', 27017)
db = client["test_mongo"]

# Eliminación de documentos de la base de datos con 1 millón de registros y todos los campos indexados

En esta sección se revisará la operación de eliminación en la colección de 1 millón de documentos indexados.

In [2]:
coll = db["extracto_cuenta_1m2"]

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [4]:
def eliminar_saldo_min(saldo_min):
    return coll.delete_many(
        {
            "saldo":
                {
                    "$lte": saldo_min
                }
        }
)

In [14]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 7 documentos
En un tiempo de 0:00:03.939482


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 3.9394 s       |

# Eliminación de documentos de la base de datos con 10 millones de registros y todos los campos indexados

En esta sección se revisará la operación de eliminación en la colección de 10 millones de documentos indexados.

In [16]:
coll = db["extracto_cuenta_10m2"]

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [17]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 111 documentos
En un tiempo de 0:00:33.027268


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 33.0276 s      |

# Eliminación de documentos de la base de datos con 30 millones de registros y todos los campos indexados

En esta sección se revisará la operación de eliminación en la colección de 1 millón de documentos ya indexados.

In [19]:
coll = db["extracto_cuenta_30m"]

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [20]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 313 documentos
En un tiempo de 0:01:40.041568


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 1 min 40.0416 s      |

# Eliminación de documentos de la base de datos con 10 millones de registros y 3 campos indexados (fecha, descripcion, nota)

En esta sección se revisará la operación de eliminación en la colección de 10 millones de documentos indexados en tres campos: fecha, descripción y nota.

In [2]:
coll = db["extracto_cuenta_10m_3idx"]

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [5]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 111 documentos
En un tiempo de 0:11:04.655072


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 11 min 4.6551 s      |

# Eliminación de documentos de la base de datos con 30 millones de registros y 3 campos indexados (fecha, descripcion, nota)

En esta sección se revisará la operación de eliminación en la colección de 30 millones de documentos indexados en tres campos: fecha, descripción y nota.

In [21]:
coll = db["extracto_cuenta_30m_3idx"]

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [22]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 313 documentos
En un tiempo de 0:27:51.043249


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 27 min 51.0432 s      |